File: AkkaRouterExample/src/main/scala/org/akka/essentials/router/custom/example/MsgEchoActor.scala
package org.akka.essentials.router.custom.example
import akka.actor.Actor
import akka.actor.ActorLogging

class MsgEchoActor extends Actor with ActorLogging{
  def receive: Receive = {
    case message =>
      log.info("Received Message {} in Actor {}", message, self.path.name)
  }
}

File: AkkaRouterExample/src/main/java/org/akka/essentials/router/example/MsgEchoActor.java
package org.akka.essentials.router.example;

import akka.actor.UntypedActor;

public class MsgEchoActor extends UntypedActor {

	@Override
	public void onReceive(Object msg) throws Exception {
		System.out.println(String.format("Received Message '%s' in Actor %s",
				msg, getSelf().path()));
	}

}
